### PR TITLE
Add Avatar widget

### PR DESCRIPTION
**Pull request by nicknisi**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] For new widgets, an entry has been added to the `.dojorc`
* [x] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**

This is part of ongoing work to improve the card implementation at #978.

The Avatar has the following properties:

- `size?: 'small' | 'medium' | 'large'` - The size of the avatar (defaults to 'medium')
- `variant?: 'square' | 'circle' | 'rounded' - The style of the avatar (defaults to 'circle')
- `secondary?: boolean` - Whether to render with secondary styles (as determined by the theme)
- `src?: string` - the path to an image file to use within the Avatar. If this is set the children are ignored
- `alt?: string` The alt text to add to the image

**Code example**
```tsx
<Avatar src="foo.png"/> // Avatar with image
<Avatar>D</Avatar> // Avatar with character text
<Avatar variant="square">D</Avatar> // square Avatar
<Avatar secondary>D</Avatar> // Avatar with secondary styling
<Avatar><Icon type="secureIcon"/></Avatar> // Avatar with Icon
```

**Avatar with image**
image

**Avatar with text**
image

**Avatar with icons**
image

**Avatar sizes**
image

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-6luu73nom.now.sh
* Tests Deployment: https://dojowidgets-g1g3ashij.now.sh

File: src/theme/material/avatar.m.css [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+.square {
+	border-radius: 0;
+}
+
+.rounded {
+	border-radius: var(--mdc-border-radius);
+}
+
+.image {
+	object-fit: cover;
```
Comment (tomdye) [inline src/theme/material/avatar.m.css]: I'm not sure that this works in ie11 or edge. How about using `background-image` instead then you would no longer need to create the img element

Comment (nicknisi) [reply src/theme/material/avatar.m.css]: Ooh, good call. I don't believe that's supported by IE11.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-ht0ub4skb.now.sh
* Tests Deployment: https://dojowidgets-jfiko0oc2.now.sh

**Pull request merged**